src/d/actor/d_a_player_boomerang.inc

```diff
@@ -16,37 +16,126 @@
 
 /* 8014956C-80149590       .text checkBoomerangAnime__9daPy_lk_cCFv */
 BOOL daPy_lk_c::checkBoomerangAnime() const {
-    /* Nonmatching */
+    return ((checkBoomerangReadyAnime()) || (checkBoomerangThrowAnime()));
 }
 
 /* 80149590-80149698       .text throwBoomerang__9daPy_lk_cFv */
 void daPy_lk_c::throwBoomerang() {
-    /* Nonmatching */
+    daBoomerang_c *boomerang = (daBoomerang_c *)mActorKeepEquip.getActor();
+
+    fopAcM_SetParam(boomerang, 1);
+    if (mpAttnActorLockOn != NULL) {
+      boomerang->setAimActor(mpAttnActorLockOn);
+    }
+    boomerang->shape_angle.y = shape_angle.y;
+    mActorKeepThrow = mActorKeepEquip;
+    mActorKeepEquip.clearData();
+    mEquipItem = daPyItem_NONE_e;
+    dComIfGp_setPlayerStatus0(0, daPyStts0_BOOMERANG_WAIT_e);
+    shape_angle.y += mBodyAngle.y;
+    current.angle.y = shape_angle.y;
+    mBodyAngle.y = 0;
+    voiceStart(0);
+    if (dComIfGp_checkPlayerStatus0(0, daPyStts0_SHIP_RIDE_e)) {
+      dComIfGp_clearPlayerStatus0(0, daPyStts0_BOOMERANG_AIM_e);
+      offModeFlg(ModeFlg_SUBJECT);
+      onModeFlg(ModeFlg_00000080);
+    } else {
+      if (mAcch.ChkGroundHit()) {
+        procWait_init();
+      }
+    }
+    return;
 }
 
 /* 80149698-801497A4       .text returnBoomerang__9daPy_lk_cFv */
-void daPy_lk_c::returnBoomerang() {
-    /* Nonmatching */
+int daPy_lk_c::returnBoomerang() {
+    dComIfGp_clearPlayerStatus0(0, daPyStts0_BOOMERANG_WAIT_e);
+    if ((((((mEquipItem == daPyItem_NONE_e) &&
+            (daPy_getPlayerActorClass() == this)) &&
+           (!dComIfGp_event_runCheck())) &&
+          (mDemo.getDemoType() == 0)) &&
+         (((m_anm_heap_upper[UPPER_MOVE2_e].mIdx == 0xffff ||
+            (checkPlayerGuard())) ||
+           ((m_anm_heap_upper[UPPER_MOVE2_e].mIdx == LKANM_BCK_DAMDASH ||
+             (checkBoomerangAnime())))))) &&
+        ((!(checkModeFlg(ModeFlg_WHIDE | ModeFlg_ROPE | ModeFlg_IN_SHIP |
+                         ModeFlg_CLIMB | ModeFlg_SWIM | ModeFlg_00080000 |
+                         ModeFlg_GRAB | ModeFlg_PUSHPULL | ModeFlg_LADDER |
+                         ModeFlg_CROUCH | ModeFlg_CRAWL)) ||
+          (mCurProc == daPyProc_SHIP_BOOMERANG_e)))) {
```

Comment:
This looks like Ghidra output, should probably be cleaned up somehow to make it a bit more readable. Maybe fix the indentation on the left to be more consistent or split this into multiple nested if statements instead of a single big one?